vortex-array/src/arrays/primitive/native_value.rs

```diff
@@ -0,0 +1,49 @@
+use vortex_dtype::{NativePType, half};
+
+/// NativeValue serves as a wrapper type to allow us to implement Hash and Eq on all primitive types.
+///
+/// Rust does not define Hash/Eq for any of the float types due to the presence of
+/// NaN and +/- 0. We don't care about storing multiple NaNs or zeros in our dictionaries,
+/// so we define simple bit-wise Hash/Eq for the Value-wrapped versions of these types.
+#[derive(Copy, Clone, Debug)]
+pub struct NativeValue<T>(pub T);
```

Comment:
We have had a need for these in 3 places. We can publicly export this wrapper to use `NativePType` semantics for native value